include/zephyr/bluetooth/conn.h

```diff
@@ -1055,6 +1057,28 @@ struct bt_conn_cb {
 	struct bt_conn_cb *_next;
 };
 
+/** @brief LTK request hook.
+ *
+ * This hook is called when the host receives a
+ * HCI_LE_Long_Term_Key_Request event.
+ *
+ * @param evt Raw event data. Can be parsed as @ref bt_hci_evt_le_ltk_request.
+ * @retval false The hook defers to the default handler.
+ * @retval true The hook takes responsibility for handling this event.
+ */
+typedef bool (*bt_conn_ltk_request_hook_t)(const uint8_t *evt);
```

Comment:
As we discussed previously in chat, the design was chosen like this to make the hook as general as possible. As offered before, the host can come with additional front-ends built on this hook.

The reason for keeping the back-end simple and general is that we have various tradeoffs to make when designing the front-end. Different application will have different needs of the implementation, which will in turn give different API capabilities and restrictions. 

Some things that have to be considered when building the front-end:
 - Is the application allowed to block in the callback? Blocking requires a resource: A thread to occupy. Which thread?
 - Will the application provide the buffer resource for constructing the HCI packet? If not, where will the buffer come from? A dedicated pool? Some global pool? What if the global pool is empty?

We can make a front-end with all the bells and whistles; its own thread, multiple callback slots and both RX and TX buffers for every connection. This will never fail and has a easy-to-use API. But it's resource hungry.

Other front-ends would be optimized, at the cost of niceness. Maybe they use global pools, and can fail to allocate and just silently drop the request. Maybe they will retry later. Maybe the application is not allowed to block, and must itself call back. etc..

What front-end requirements do you have in mind? Let's discuss this and I'll see what I can provide.

Reply:
I think we can keep the callback called from the same context.  We can also keep using the same buffer poll.
I was only thinking about slightly simplifying user API by modifying function arguments and keeping buffer allocations on Bluetooth stack side. Thanks to that application's job would be only to fill the buffer with LTK if needed.
Still, since the API would not probably be used for most of the typical use-cases, I would not insist on simplifying the API (it's more like improvement idea from my end).